internal/security/secretstore/vmkencryption.go

```diff
@@ -159,40 +158,38 @@ func (v *VMKEncryption) DecryptInitResponse(initResp *secretstoreclient.InitResp
 		hexNonce := initResp.Nonces[i]
 		nonce, err := hex.DecodeString(hexNonce)
 		if err != nil {
-			return fmt.Errorf("failed to decode hex bytes of nonce: %w", err)
+			return decryptedResponse, fmt.Errorf("failed to decode hex bytes of nonce: %w", err)
 		}
 
 		cipherText, err := hex.DecodeString(hexCiphertext)
 		if err != nil {
-			return fmt.Errorf("failed to decode hex bytes of ciphertext: %w", err)
+			return decryptedResponse, fmt.Errorf("failed to decode hex bytes of ciphertext: %w", err)
 		}
 
-		keyShare, err := v.gcmDecryptKeyshare(cipherText, nonce, i) // Unwrap using a unique AES key
+		keyShare, err := v.gcmDecryptKeyShare(cipherText, nonce, i) // Unwrap using a unique AES key
 		if err != nil {
-			return fmt.Errorf("failed to unwrap key %d: %w", i, err)
+			return decryptedResponse, fmt.Errorf("failed to unwrap key %d: %w", i, err)
 		}
 
 		newKeys[i] = hex.EncodeToString(keyShare)
 		newKeysBase64[i] = base64.StdEncoding.EncodeToString(keyShare)
 	}
 
-	initResp.Keys = newKeys
-	initResp.KeysBase64 = newKeysBase64
-	initResp.EncryptedKeys = nil
-	initResp.Nonces = nil
-	return nil
+	decryptedResponse.Keys = newKeys
+	decryptedResponse.KeysBase64 = newKeysBase64
+	return decryptedResponse, nil
```

Comment:
@lenny-intel Please copy input response to decrypted response and only modify the four affected fields.

Reply:
@bnevis-i , why are the inputs needed in the decrypted response?  They are not used after decrypting.

Reply:
K, got it. The root token is needs to stay.

Reply:
Now it makes more sense on why passing reference. Restoring to that approach.

Reply:
Restore original approach